test/lib/jdk/test/lib/net/SimpleHttpServer.java

```diff
@@ -99,10 +96,12 @@ public void handle(final HttpExchange t) throws IOException {
                 Headers rMap = t.getResponseHeaders();
                 try (OutputStream os = t.getResponseBody()) {
                     URI uri = t.getRequestURI();
-                    String path = uri.getPath();
+                    String path = uri.getRawPath();
+                    assert path.isEmpty() || path.startsWith("/");
                     Path fPath;
                     try {
-                        fPath = Paths.get(docRoot, path);
+                        uri = URI.create(rootUri.getRawPath() + path).normalize();
+                        fPath = Path.of(uri.getRawPath());
```

Comment:
This is wrong. It should be `fpath = Path.of(uri);`. See https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/nio/file/Path.html#of(java.net.URI)